pkg/arvo/app/btc-node-hook.hoon

```diff
@@ -0,0 +1,373 @@
+::  btc-node-hook: send JSON rpc requests to bitcoin full node
+::  and poke the responses into the btc-node-store
+::
+/-  *btc-node-hook, *btc-node-store, sole
+/+  default-agent, sole, lib=btc-node-json, verb
+::
+=>  |%
+    +$  card  card:agent:gall
+    +$  versioned-state
+      $%  [%0 state-zero]
+      ==
+    ::
+    +$  state-zero
+      $:  endpoint=@t
+          headers=header-list:http
+      ==
+    --
+::
+=|  state-zero
+=*  state  -
+::  Main
+::
+%+  verb  |
+^-  agent:gall
+=<  |_  =bowl:gall
+    +*  this      .
+        btc-core  +>
+        bc        ~(. btc-core bowl)
+        def       ~(. (default-agent this %|) bowl)
+    ::
+    ++  on-init
+      ^-  (quip card _this)
+      :-  ~
+      %_  this
+        endpoint  ''
+        headers   ~
+      ==
+    ::
+    ++  on-save   !>(state)
+    ++  on-load
+      |=  old=vase
+      `this(state !<(state-zero old))
+    ::
+    ++  on-poke
+      |=  [=mark =vase]
+      ^-  (quip card _this)
+      |^  ?+    mark    (on-poke:def mark vase)
+              %btc-node-hook-action
+            (hook-action !<(btc-node-hook-action vase))
+          ::
+              %btc-node-hook-command
+            (hook-command !<(btc-node-hook-command vase))
+          ==
+      ::
+      ++  hook-action
+        |=  act=btc-node-hook-action
+        ^-  (quip card _this)
+        =/  body=request:rpc:jstd
+          (request-to-rpc:btc-rpc:lib act)
+        =/  req=request:http
+          :*  %'POST'
+              (endpoint-url:bc act)
+              headers
+              =,  html
+              %-  some
+                %-  as-octt:mimes
+                  (en-json (request-to-json:rpc:jstd body))
+          ==
+        =/  out  *outbound-config:iris
+        :_  this
+        [%pass /[(scot %da now.bowl)] %arvo %i %request req out]~
+      ::
+      ++  hook-command
+        |=  comm=btc-node-hook-command
+        ^-  (quip card _this)
+        ?+    -.comm  ~|  [%unsupported-hook-command -.comm]  !!
+            %credentials
+          :_  this(endpoint url.comm, headers heads.comm)
+          [%pass / %arvo %d %flog [%text "credentials updated..."]]~
+        ==
+      --
+    ::
+    ++  on-watch  on-watch:def
+    ++  on-leave  on-leave:def
+    ++  on-peek   on-peek:def
+    ++  on-agent  on-agent:def
+    ++  on-arvo
+      |=  [=wire =sign-arvo]
+      ^-  (quip card _this)
+      =*  response  client-response.sign-arvo
+      =^  cards  state
+        ?+    +<.sign-arvo    (on-arvo:def wire sign-arvo)
+            %http-response    (http-response:bc wire response)
+        ==
+      [cards this]
+    ::
+    ++  on-fail   on-fail:def
+    --
+::
+|_  =bowl:gall
+::
+++  httr-to-rpc-response
+  |=  hit=httr:eyre
+  ^-  response:rpc:jstd
+  ~|  hit
+  :: ?.  ?=(%2 (div p.hit 100))
+  ::   fail+hit
```

Comment:
This should probably be `=(2 (div p.hit 100))`. Do we not want to enable this for some reason?

Reply:
Re-adding the handle-error (called now `parse-error`) here instead so we get a better error message without http headers.